DQM/SiStripMonitorClient/plugins/SiStripBadComponentInfo.h

```diff
@@ -0,0 +1,75 @@
+#ifndef SiStripMonitorClient_SiStripBadComponentInfo_h
+#define SiStripMonitorClient_SiStripBadComponentInfo_h
+// -*- C++ -*-
+//
+// Package:     SiStripMonitorClient
+// Class  :     SiStripBadComponentInfo
+// 
+/**\class SiStripBadComponentInfo SiStripBadComponentInfo.h DQM/SiStripMonitorCluster/interface/SiStripBadComponentInfo.h
+
+ Description: 
+      Checks the # of SiStrip FEDs from DAQ
+ Usage:
+    <usage>
+
+*/
+//
+//          Author:  Suchandra Dutta
+//         Created:  Fri Jan 26 10:00:00 CET 2018
+//
+
+#include <string>
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DQMServices/Core/interface/DQMEDHarvester.h"
+
+#include "CalibFormats/SiStripObjects/interface/SiStripQuality.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <map>
+
+class DQMStore;
+class MonitorElement;
+
+class SiStripBadComponentInfo: public DQMEDHarvester {
+
+ public:
+
+  /// Constructor
+  SiStripBadComponentInfo(const edm::ParameterSet& ps);  
+  ~SiStripBadComponentInfo() override;
+
+protected:
+
+  void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  void dqmEndJob(DQMStore::IBooker &, DQMStore::IGetter &) override;  //performed in the endJob
+
+private:
+  void checkBadComponents();
+  void bookBadComponentHistos(DQMStore::IBooker &ibooker, DQMStore::IGetter &igetter);
+  void fillBadComponentHistos(int xbin, int component,SiStripQuality::BadComponent& BC);
+  void createSummary(MonitorElement* me);
+
+  DQMStore* dqmStore_;
```

Comment:
It is forbidden to have a pointer to DQMStore, please remove it.

Reply:
for my curiosity: if using DQMStore pointers is forbidden, couldn't it be enforced by making this impossible, like not being able to compile because it is not a know type?